Content.Server/Box/BoxSystem.cs

```diff
@@ -0,0 +1,75 @@
+﻿using System.Linq;
+using Content.Server.Box.Components;
+using Content.Server.Storage.Components;
+using Content.Shared.Box;
+using Content.Shared.Box.Components;
+using Content.Shared.Movement.Components;
+using Robust.Shared.Player;
+
+namespace Content.Server.Box;
+
+public sealed class BoxSystem : SharedBoxSystem
+{
+    [Dependency] private readonly SharedAudioSystem _audio = default!;
+    public override void Initialize()
+    {
+        base.Initialize();
+        SubscribeLocalEvent<BoxComponent, StorageBeforeCloseEvent>(OnBeforeStorageClosed);
+        SubscribeLocalEvent<BoxComponent, StorageAfterOpenEvent>(AfterStorageOpen);
+    }
+
+    public override void Update(float frameTime)
+    {
+        base.Update(frameTime);
+
+        foreach (var box in EntityQuery<BoxComponent>())
+        {
+            if (box.EffectPlayed)
+                box.Accumulator += frameTime;
+
+            if (box.Accumulator >= box.EffectCooldown)
+            {
+                box.Accumulator -= box.EffectCooldown;
+                box.EffectPlayed = false;
+            }
+        }
+    }
+
+    private void OnBeforeStorageClosed(EntityUid uid, BoxComponent component, StorageBeforeCloseEvent args)
+    {
+        //Grab the first mob in the hash to set as the mover and to prevent other mobs from entering.
+        var firstMob = args.Contents.Where(e => HasComp<MobMoverComponent>(e)).Select(e => e).FirstOrDefault();
+
+        //Set the movement relay for the box as the first mob
+        if (component.Mover == null && args.Contents.Contains(firstMob))
+        {
+            var relay = EnsureComp<RelayInputMoverComponent>(firstMob);
+            relay.RelayEntity = uid;
+            component.Mover = firstMob;
+        }
+
+        //Check the contents of the box and remove any mobs other than the driver.
+        foreach (var entity in args.Contents)
+        {
+            if (HasComp<MobMoverComponent>(entity) && firstMob != entity)
+                args.Contents.Remove(entity);
+        }
```

Comment:
- `Select(e => e)` is redundant
- Pretty sure the foreach loop will cause modified-while-enumerating errors
- #11692 means the relay input stuff needs to be updated
- Instead of effectively doing HasComp twice, maybe just do `args.Contents.Where(e => HasComp<MobMoverComponent>(e)).ToList()` once?